submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/validator/filelist/FileListValidator.kt

```diff
@@ -35,17 +36,23 @@ class FileListValidator(
             preferredSources = emptyList()
         )
         val fileSources = fileSourcesService.submissionSources(fileSourcesRequest)
+        validateFileList(fileListName, fileSources)
+    }
+
+    fun validateFileList(fileListName: String, fileSources: FileSourcesList) {
         val fileListFile = getFileListFile(fileListName, fileSources)
 
-        fileListFile.inputStream().use { validateFileList(it, SubFormat.fromFile(fileListFile), fileSources) }
+        fileListFile.inputStream().use {
+            validateFileList(fileListName, it, SubFormat.fromFile(fileListFile), fileSources)
+        }
     }
 
-    private fun validateFileList(stream: InputStream, format: SubFormat, filesSource: FileSourcesList) {
-        serializationService
-            .deserializeFileList(stream, format)
+    private fun validateFileList(name: String, stream: InputStream, format: SubFormat, filesSource: FileSourcesList) {
+        val fileListFiles = serializationService.deserializeFileList(stream, format).take(fileListLimit).toList()
```

Comment:
I think this is incorrectly changing how method use to work, before we were checking for the first `n` null files and reporting them, now we are checking if the first `n` has a empty file. Which means i.e the file 1001 does not exists this method will not fail. Do I am missing something?